pkg/controllers/generic/import_syncer.go

```diff
@@ -365,6 +401,11 @@ func (s *importer) updateVirtualAnnotations(a map[string]string) map[string]stri
 }
 
 func (s *importer) addAnnotationsToPhysicalObject(ctx *synccontext.SyncContext, pObj client.Object) error {
+	if s.syncerOptions.IsClusterScopedCRD {
+		// do not add annotations to physical object
+		return nil
+	}
```

Comment:
nit-pick: would be better to check this before calling this function, and leave the function concerned with just adding the annotations

Reply:
actually this function is getting called from more than 1 places, at-least 2 if I recall correctly
hence for the sake of simplicity I put this check here. But if you strongly feel we should do it the other way I can change it

Reply:
I don't feel that strongly about this :)